From 6fe6a176eaa6c1e8be572c398dc7d0621a4ebe72 Mon Sep 17 00:00:00 2001
From: "zhi.zhou" <zhi.zhou@amlogic.com>
Date: Fri, 3 Jan 2014 20:25:56 +0800
Subject: [PATCH 2711/5965] pd#85686 tsyc bug for sohu,when streambuf is
 big(15M),we need about 6K pts for video.and a bug on too many av-discontinue
 may timeout bug.

---
 drivers/amlogic/amports/ptsserv.c | 2 +-
 drivers/amlogic/amports/tsync.c   | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/amports/ptsserv.c b/drivers/amlogic/amports/ptsserv.c
index 31dfc331252a..4857cdeea302 100755
--- a/drivers/amlogic/amports/ptsserv.c
+++ b/drivers/amlogic/amports/ptsserv.c
@@ -10,7 +10,7 @@
 
 #include "vdec_reg.h"
 
-#define VIDEO_REC_SIZE  4096
+#define VIDEO_REC_SIZE  8192
 #define AUDIO_REC_SIZE  8192
 #define VIDEO_LOOKUP_RESOLUTION 2500
 #define AUDIO_LOOKUP_RESOLUTION 1024
diff --git a/drivers/amlogic/amports/tsync.c b/drivers/amlogic/amports/tsync.c
index d4d632237ae5..a25f6798a4b1 100755
--- a/drivers/amlogic/amports/tsync.c
+++ b/drivers/amlogic/amports/tsync.c
@@ -376,7 +376,7 @@ static int tsync_mode_switch(int mode,unsigned long diff_pts,int jump_pts)
 	int old_tsync_mode=tsync_mode;
 	int old_tsync_av_mode=tsync_av_mode;
 	char VA[]="VA--";
-       unsigned int olddur=tsync_av_dynamic_duration_ms;
+       unsigned int oldtimeout=tsync_av_dynamic_timeout_ms;
 	
 	printk("%c-discontinue,pcr=%d,vpts=%d,apts=%d,diff_pts=%lu,jump_Pts=%d\n",mode,timestamp_pcrscr_get(),timestamp_vpts_get(),timestamp_apts_get(),diff_pts,jump_pts);
 	if (!tsync_enable) {
@@ -452,7 +452,7 @@ static int tsync_mode_switch(int mode,unsigned long diff_pts,int jump_pts)
 
 
 	
-	if(olddur!=tsync_av_dynamic_duration_ms){/*duration changed,update new timeout.*/
+	if(oldtimeout!=tsync_av_latest_switch_time_ms+tsync_av_dynamic_duration_ms){/*duration changed,update new timeout.*/
 			tsync_av_dynamic_timeout_ms=tsync_av_latest_switch_time_ms+tsync_av_dynamic_duration_ms;
 	}
 	printk("discontinue-tsync_mode:%c->%c,state:%c->%c,debugcnt=0x%x,diff_pts=%lu,tsync_mode=%d\n",
-- 
2.19.0

